README.md

```diff
@@ -0,0 +1,83 @@
+# Sign In With Ethereum
+
+This crate provides a pure Rust implementation of EIP-4631: Sign In With Ethereum.
+
+## Installation
+
+SIWE can be easily installed in any Rust project by including it in said project's `cargo.toml` file:
+
+``` toml
+siwe = "0.1"
+```
+
+## Usage
+
+SIWE exposes a `Message` struct which implements EIP-4361.
+
+### Parsing a SIWE Message
+
+Parsing is done via the `Message` implementation of `FromStr`:
+
+``` rust
+let message: Message = string_message.parse()?;
+```
+
+### Verifying and Authenticating a SIWE Message
+
+Verification and Authentication is performed via EIP-191, using the `.address` field of the `Message` as the expected signer:
+
+``` rust
+message.verify_eip191(&signature)?;
```

Comment:
I think the validity check should be done by default in this function, maybe provide a flag/parameter to enable/disable it.

Reply:
I impl'd a new `.verify(sig: &[u8; 65])` fn to do combined verification, as well as a `valid_at` fn in case thats useful, and updated the readme to reflect